### PR TITLE
feat(gallery): grid templates - FRONT-4622

### PR DESCRIPTION
**Pull request by emeryro**

### CODE REVIEW

File: src/specs/components/gallery/demo/data.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = {
       // Image with thumbnail
       thumbnail: {
         img: {
-          src: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg',
+          src: 'https://commission.europa.eu/sites/default/files/styles/oe_theme_medium_no_crop/avportal/P-035170/00-08.jpg?',
```
Comment (planctus) [inline src/specs/components/gallery/demo/data.js]: are we going to use a file from the EC production website..?

File: src/implementations/twig/components/gallery/gallery.html.twig [hunk 2]
```diff
@@ -80,6 +82,10 @@
 {% if _grid %}
   {% set _css_class = _css_class ~ ' ecl-gallery--grid' %}
 
+  {% if _grid_template is not empty %}
+    {% set _css_class = _css_class ~ ' ecl-gallery--template-' ~ _grid_template %}
+  {% endif %}
+
   {% if _column is not empty %}
```
Comment (planctus) [inline src/implementations/twig/components/gallery/gallery.html.twig]: this selector is irrelevant when using a "grid template" right? If i'm correct better not setting in those cases and also in the story, if possible, would be less confusing if we would hide the control for the number of columns

**Review by planctus — COMMENTED** [review]
I guess that the feeling is more or less the same about this, this overengineering of the gallery component doesn't seem to come to an end. Anyway, the different layouts work well, i just left a couple of comments. Thanks

File: src/implementations/twig/components/gallery/gallery.html.twig (hunk 2 above)
Comment (emeryro) [reply src/implementations/twig/components/gallery/gallery.html.twig]: In fact it is used only when using the "default" grid template (a plain grid without any modification). I'll add a test for that in the twig template, and check the story

**Pull request merged**